source/overloading.tex

```diff
@@ -1571,23 +1571,49 @@
 with the assumption that any dependent base class has
 no virtual functions and no virtual base classes, and
 the initializer is a non-empty \grammarterm{braced-init-list} or
-parenthesized \grammarterm{expression-list},
+parenthesized \grammarterm{expression-list}, and
+there are no \grammarterm{deduction-guide}{s} for \tcode{C},
 the set contains an additional function template,
 called the \defnadj{aggregate deduction}{candidate}, defined as follows.
 Let $x_1, \dotsc, x_n$ be the elements
 of the \grammarterm{initializer-list} or
 \grammarterm{designated-initializer-list}
 of the \grammarterm{braced-init-list}, or
 of the \grammarterm{expression-list}.
-For each $x_i$, let $e_i$ be the corresponding element
+For each $x_i$, let $e_i$ be the corresponding aggregate element
 of \tcode{C} or of one of its (possibly recursive) subaggregates
-that would be initialized by $x_i$\iref{dcl.init.aggr}
-if brace elision is not considered for any element
-that has a dependent type.
-If there is no such element $e_i$, the program is ill-formed.
+that would be initialized by $x_i$\iref{dcl.init.aggr} if
+\begin{itemize}
+\item
+brace elision is not considered for any aggregate element
+that has a dependent non-array type or
+an array type with a value-dependent bound, and
+\item
+each non-trailing aggregate element that is a pack expansion
+is assumed to correspond to no elements of the initializer list, and
+\item
+a trailing aggregate element that is a pack expansion is assumed to correspond
+to all remaining elements of the initializer list (if any).
+\end{itemize}
+If there is no such aggregate element $e_i$ for any $x_i$,
+the aggregate deduction candidate is not added to the set.
 The aggregate deduction candidate is derived as above
 from a hypothetical constructor $\tcode{C}(\tcode{T}_1, \dotsc, \tcode{T}_n)$,
-where $\tcode{T}_i$ is the declared type of the element $e_i$.
+where
+\begin{itemize}
+\item
+if $e_i$ is of array type and
+$x_i$ is a \grammarterm{braced-init-list} or string literal,
+$T_i$ is an rvalue reference to the declared type of $e_i$, and
```

Comment:
T should not be in math font, use $\tcode{T}_i$ instead

Reply:
Fixed.